siriuspy/siriuspy/pwrsupply/pru.py

```diff
@@ -212,11 +214,16 @@ def get_connected(self, ID_device):
         """Return connected state of ID_device."""
         return self._connected[ID_device]
 
+    def get_wfmdata(self, ID_device):
+        """Return wfmdata of a given ID_device."""
+        return self._waveforms[ID_device]
+
     def set_wfmdata(self, ID_device, wfmdata):
         """Set waveform of a device."""
         sorted_IDs = sorted(self._waveforms.keys())
         if ID_device not in sorted_IDs:
-            raise ValueError
+            print('ID_device {} not defined!'.format(ID_device))
+            return
         else:
             self._waveforms[ID_device] = wfmdata[:]
```

Comment:
what if wfmdata do not have the right length?
If its length is smaller than 4000, all values will be overwritten, right?
If its length is larger than 4000, the resulting waveform will also be larger.

Reply:
PRU library deals with variable-size wavelengths. PRU._waveforms is passed as argument to a function in this library. Since class PRU represents PRU and its library, I decided not to impose policies on wfmdata in it, just like the PRU library. The other classes that uses PRU do proper checkings. (model.PowerSupply, in this case)